packages/agreement-process/test/agreementService.test.ts

```diff
@@ -0,0 +1,790 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable sonarjs/no-identical-functions */
+/* eslint-disable functional/immutable-data */
+import { describe, expect, it } from "vitest";
+import {
+  Agreement,
+  Descriptor,
+  EService,
+  EServiceAttribute,
+  Tenant,
+  TenantAttribute,
+  agreementCreationConflictingStates,
+  agreementState,
+  descriptorState,
+  generateId,
+  tenantAttributeType,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { generateMock } from "@anatine/zod-mock";
+import { AuthData } from "pagopa-interop-commons";
+import { AgreementQuery } from "../src/services/readmodel/agreementQuery.js";
+import { TenantQuery } from "../src/services/readmodel/tenantQuery.js";
+import { EserviceQuery } from "../src/services/readmodel/eserviceQuery.js";
+import { createAgreementLogic } from "../src/services/agreementCreationProcessor.js";
+import { ApiAgreementPayload } from "../src/model/types.js";
+import { toAgreementStateV1 } from "../src/model/domain/toEvent.js";
+import {
+  agreementAlreadyExists,
+  descriptorNotInExpectedState,
+  eServiceNotFound,
+  missingCertifiedAttributesError,
+  notLatestEServiceDescriptor,
+  tenantIdNotFound,
+} from "../src/model/domain/errors.js";
+import { AgreementQueryFilters } from "../src/services/readmodel/readModelService.js";
+import { expectPastTimestamp, randomArrayItem } from "./utils/utils.js";
+
+export const notDraftDescriptorStates = Object.values(descriptorState).filter(
+  (state) => state !== descriptorState.draft
+);
+
+describe("AgreementService", () => {
+  describe("createAgreement (success cases)", () => {
+    it("should create an Agreement when EService producer and Agreement consumer are the same Tenant", async () => {
+      const tenant: Tenant = generateMock(Tenant);
+      const descriptor: Descriptor = {
+        ...generateMock(Descriptor),
+        state: descriptorState.published,
+      };
+      const eservice: EService = {
+        ...generateMock(EService),
+        producerId: tenant.id,
+        descriptors: [descriptor],
+      };
+
+      const agreementQueryMock = {
+        getAllAgreements: () => Promise.resolve([]),
+      } as unknown as AgreementQuery;
```

Comment:
a simple `as AgreementQuery` doesn't work?

Reply:
Doesn't work 🥲 
<img width="963" alt="image" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/9c9fc984-1207-4ba0-b94b-633fe96207c5">